File: AndroidBasics/Intent/app/src/main/java/com/devby/multipleactivities/MainActivity.java
package com.devby.multipleactivities;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {
    String userName;
    EditText editText;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        editText = findViewById(R.id.editText);
        userName = "";
    }
    public void changeActivity(View view){
        userName = editText.getText().toString();
        Intent intent = new Intent(MainActivity.this,SecondActivity.class);
        intent.putExtra("UserInput",userName);

        startActivity(intent);




    }
}

File: AndroidBasics/Lifecycle/app/src/main/java/com/devby/lifecycle/MainActivity.kt
package com.devby.lifecycle

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.devby.lifecycle.databinding.ActivityMainBinding
//LifeCycle,EditText,Intent,Context,Alert
class MainActivity : AppCompatActivity() {
    private lateinit var binding : ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        binding = ActivityMainBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
        println("onCreate() called")
    }

    override fun onStart() {
        super.onStart()
        println("onStart() called")
    }

    override fun onResume() {
        super.onResume()
        println("onResume() called")
    }

    override fun onPause() {
        super.onPause()
        println("onPause() called")
    }
    override fun onStop() {
        super.onStop()
        println("onStop() called")
    }

    override fun onDestroy() {
        super.onDestroy()
        println("onDestroy()  called")
    }
    fun sonrakiSayfa(view : View){
        val userInput = binding.editText.text.toString()
        binding.textView.text = "İsim: ${userInput}"

        //intent
        val intent = Intent(this,SecondActivity::class.java)
        intent.putExtra("isim",userInput)
        startActivity(intent)
            //finish()






    }




}